mat64/gsvd.go

```diff
@@ -0,0 +1,296 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+// Based on the SingularValueDecomposition class from Jama 1.0.3.
+
+package mat64
+
+import (
+	"github.com/gonum/blas/blas64"
+	"github.com/gonum/floats"
+	"github.com/gonum/lapack"
+	"github.com/gonum/lapack/lapack64"
+	"github.com/gonum/matrix"
+)
+
+// GSVD is a type for creating and using the Generalized Singular Value Decomposition
```

Comment:
My interest is from [this paper](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0030098), but there are other uses. I don't think we have put more information on other things (for example SVD) than the properties defined by the equations.